gui/default/syncthing/folder/ignoresService.js

```diff
@@ -0,0 +1,257 @@
+                return oldPattern;
+            }
+            return null;
+        };
+
+        self.matchingPattern = function(file) {
+            return self.data.patterns.find(function(pattern) {
+                // Only consider patterns that match a simple path
+                if (!pattern.isSimple) return false;
+
+                var absPath = '/' + file.path;
+                return pattern.matchFunc(absPath);
+            });
+        }
+
+        /*
+         * private definitions
+         */
+
+        function getIgnores(folderId) {
+            return $http.get('rest/db/ignores', {
+                params: { folder: folderId }
+            }).then(function (response) {
+                var data = response.data;
+                return (data.ignore || []).map(parsePattern);
+            });
+        }
+
+        function parsePattern(line) {
+            var stripResult = splitPrefix(line.trim());
+            var prefixes = stripResult[0];
+            var hasPrefix = prefixes['(?i)'] || prefixes['(?d)'];
+            var path = toPath(stripResult[1]);
+            var matchFunc = !hasPrefix && chooseMatcher(path);
+
+            return {
+                text: line,
+                isSimple: !!matchFunc,
+                isNegated: prefixes['!'],
+                path: path,
+                matchFunc: matchFunc,
+            };
+        };
+
+        // Adapted from lib/ignore/ignore.go@parseLine
+        function splitPrefix(line) {
+            var seenPrefix = { '!': false, '(?i)': false, '(?d)': false };
+
+            var found = true;
+            while (found === true) {
+                found = false;
+                for (var prefix in seenPrefix) {
+                    if (!seenPrefix[prefix] && line.indexOf(prefix) === 0) {
+                        seenPrefix[prefix] = true;
+                        line = line.slice(prefix.length);
+                        found = true;
+                        break;
+                    }
+                }
+            }
+
+            return [seenPrefix, line];
+        }
+
+        // Infer a path from wildcards that might be used to match a file path
+        // by prefix (for simple patterns)
+        // The changes made to the path make it more specific than the original
+        // pattern. It is useful for prefix matcher. Should a glob matcher be
+        // added this path should not be used to match because it does not
+        // represent the complete pattern. The most common effect of the changes
+        // is matching patterns like `*`, `**`, or `/*` without a special case
+        function toPath(line) {
+            // Add a leading slash when the pattern begins with wildcards
+            // because a pattern beginning with * or ** will match entries at
+            // the root (as well as those in child directories)
+            line = line.replace(/^\*+/, '/$&'); // `$&` inserts the matched substring
```

Comment:
If there are still `*`s after trimming trailing ones on :193, then yes it will be ignored in chooseMatcher. Practically speaking, :189 is necessary to turn pattern `*` into `/*` so it is interpreted as the path `/` and matches all files.